operator/cmd/mesh/install.go

```diff
@@ -318,5 +321,16 @@ func getProfileAndEnabledComponents(setOverlay []string, inFilenames []string, f
 			}
 		}
 	}
-	return profile, enabledComponents, nil
+
+	// check if non default namespace is used
+	globalI := iop.Spec.Values["global"]
+	global, ok := globalI.(map[string]interface{})
+	if !ok {
+		return profile, controller.IstioNamespace, enabledComponents, nil
+	}
+	ns, ok := global["istioNamespace"].(string)
+	if !ok {
+		return profile, controller.IstioNamespace, enabledComponents, nil
+	}
+	return profile, ns, enabledComponents, nil
```

Comment:
@linsun I think you can reuse the function: https://github.com/istio/istio/blob/cbe5f1108f10ba3803e5ac3a6ae1a8e35e26531c/operator/pkg/apis/istio/v1alpha1/common.go#L30

Reply:
Ah thanks @morvencao!